packages/ui-library/src/components/sidebar-navigation/link.js

```diff
@@ -0,0 +1,48 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useCallback } from "react";
+import { useNavigationContext } from "./index";
+
+/**
+ * @param {JSX.ElementClass} [as="a"] The component.
+ * @param {string} [pathProp="href"] The key of the path in the props.
+ * @param {?JSX.node} [children=null] The content.
+ * @param {string} [className] The classname.
+ * @param {Object} [props] Extra props.
+ * @returns {JSX.Element} The element.
+ */
+export const Link = ( { as: Component = "a", pathProp = "href", children = null, className, ...props } ) => {
```

Comment:
Children default is null, same as above.
Why onClick is missing here?

Reply:
> Why onClick is missing here?

It is not necessarily a part of the API of the link itself. The `onClick` is not a thing for the default `as` here (anchor).
However, we do interact with `onClick` because we use it to close the Mobile menu. And we still want to make it possible for people to use it. So we ensure to call it if it is a prop.

We do a similar thing with the `props?.[ pathProp ]`, even though there we actually have no way of knowing what it is 😁